### DISCUSSION TITLE
How can I check the collision between an ellipsoid and a point cloud fast

### DISCUSSION THREAD

**Discussion by lwygzh** [coal-library/coal#541] [General]
Hi,

I noticed that you can use point cloud as the collision object. 

However, it requires us to save the point cloud as a `.obj`, `.stl`, or `.dae` file in order to process it. 

I do not think the process of saving points clouds into these file types and then load them is computationally efficient enough for online collision detection especially when the point cloud is changing, is that correct? Is there any way to get around this?

**Comment by florent-lamiraux** [comment]
Hello @lwygzh .
If you look into the tests:
https://github.com/humanoid-path-planner/hpp-fcl/blob/16f250120375e50ed232ac873c9b87d9570e7614/test/bvh_models.cpp#L58
you will find an example of construction of point cloud without reading a file.
A PointCloud is a BVHModel without triangles.
